parsl/jobs/states.py

```diff
@@ -0,0 +1,116 @@
+import os
+from enum import IntEnum
+import logging
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+class JobState(IntEnum):
+    """Defines a set of states that a job can be in"""
+    UNKNOWN = 0
+    PENDING = 1
+    RUNNING = 2
+    CANCELLED = 3
+    COMPLETED = 4
+    FAILED = 5
+    TIMEOUT = 6
+    HELD = 7
+
+    def __str__(self) -> str:
+        return self.__class__.__name__ + "." + self.name
```

Comment:
This is a bit different from the repr changes that follow on from #1966 style repr tidyups:

repr on this class is still the default, which gives something like this (on Python 3.11):

```
>>> repr(p.JobState.COMPLETED)
'<JobState.COMPLETED: 4>'
```

but this `__str__` was addred in PR #2653 specifically because the str behaviour changed from Python 3.10 to Python 3.11 to rendering an IntEnum as the integer value, rather than as the name of the state.

This makes sense for the use-case it was intended for (when tightening an already existing `int` field into an `IntEnum`) but in the Parsl case, I wanted to preserve the loose human representation being the state name (which is the case in Python <=3.10)

For example, when used in log messages, it is nicer to represent the state to a human reader as `"COMPLETED"` rather than as `4`.